### PR TITLE
Fix account not found when doing local file editing.

### PR DESCRIPTION
**Pull request by allexzander**
Signed-off-by: alex-z <blackslayer4@gmail.com>

### CODE REVIEW

File: src/gui/accountmanager.cpp [hunk 1]
```diff
@@ -348,6 +348,18 @@ AccountStatePtr AccountManager::account(const QString &name)
     return it != _accounts.cend() ? *it : AccountStatePtr();
 }
 
+AccountStatePtr AccountManager::accountFromDavUserAndHost(const QString &davUserNameAndHost)
+{
+    const auto davUserNameAndHostSplit = davUserNameAndHost.split(QLatin1Char('@'));
+    if (davUserNameAndHostSplit.size() < 2) {
+        return AccountStatePtr();
+    }
+    const auto it = std::find_if(_accounts.cbegin(), _accounts.cend(), [davUserNameAndHostSplit](const auto &acc) {
```
Comment (claucambra) [inline src/gui/accountmanager.cpp:357]:
```suggestion
    const auto it = std::find_if(_accounts.cbegin(), _accounts.cend(), [&davUserNameAndHostSplit](const auto &acc) {
```

**Review by claucambra — APPROVED** [review]

**Review by mgallien — COMMENTED** [review]
partial review pending more thorough review

File: src/libsync/account.cpp [hunk 2]
```diff
@@ -147,7 +147,21 @@ void Account::setAvatar(const QImage &img)
 
 QString Account::displayName() const
 {
-    QString dn = QString("%1@%2").arg(credentials()->user(), _url.host());
+    const auto userName = [this]() {
+        if (!_davUser.isEmpty()) {
+            return _davUser;
```
Comment (mgallien) [inline src/libsync/account.cpp:152]: this is wrong for something that will be a display name
`_davUser` is the user backend internal id coming from server and should never be display to teh user

**Review by mgallien — CHANGES_REQUESTED** [review]
the changes to `Account::displayName()` are wrong
display name has nothing to do with the user id stored in `_davUser`

**Review by mgallien — APPROVED** [review]
thanks

**Pull request merged**